test/test-manager/src/tests/tunnel.rs

```diff
@@ -203,6 +203,49 @@ pub async fn test_udp2tcp_tunnel(
     Ok(())
 }
 
+/// Use Shadowsocks obfuscation. This tests whether the daemon can establish a Shadowsocks tunnel.
+/// Note that this doesn't verify that Shadowsocks is in fact being used.
+#[test_function]
+pub async fn test_wireguard_over_shadowsocks(
+    _: TestContext,
+    rpc: ServiceClient,
+    mut mullvad_client: MullvadProxyClient,
+) -> anyhow::Result<()> {
+    mullvad_client
+        .set_obfuscation_settings(relay_constraints::ObfuscationSettings {
+            selected_obfuscation: SelectedObfuscation::Shadowsocks,
+            shadowsocks: ShadowsocksSettings {
+                port: Constraint::Any,
+            },
+            ..Default::default()
+        })
+        .await
+        .context("Failed to enable shadowsocks")?;
+
+    let relay_settings = RelaySettings::Normal(RelayConstraints {
+        tunnel_protocol: Constraint::Only(TunnelType::Wireguard),
+        ..Default::default()
+    });
+
+    set_relay_settings(&mut mullvad_client, relay_settings)
+        .await
+        .context("Failed to update relay settings")?;
+
```

Comment:
Calling `set_obfuscation_settings` separetly from `set_relay_settings` is not necessary, right? We could also make use of the relay query builder here, which is nice if we change the underlying settings format

``suggestion
    set_relay_settings(
        &mut mullvad_client,
        RelaySettings::Normal(
            RelayQueryBuilder::new()
                .wireguard()
                .shadowsocks()
                .into_constraint(),
        ),
    )
    .await?;
```

Reply:
It is necessary to call `set_obfuscation_settings`. When the query is converted to a `RelaySettings`, it will not use any of the obfuscation constraints, as the relay settings struct doesn't contain any obfuscation constraints.

This is obviously a footgun, but that's how it works for the time being. Maybe our helper function `set_relay_settings` should take care of this? That doesn't solve the underlying problem, though.

Reply:
Maybe instead of implementing `From<RelayQuery>` for `RelayConstraints`, we should add a method to `RelayQuery` that returns obfuscation settings (if there are any) as well as relay settings. That would force the caller to pay attention to both.